http/grafana.go

```diff
@@ -0,0 +1,288 @@
+	var retval simpleJsonRequestInterface
+	switch method {
+	case "query":
+		retval = &simpleJsonQueryRequest{Backend: defaultBackend}
+		break
+	case "search":
+		retval = &simpleJsonSearchRequest{simpleJsonQueryRequest{Backend: defaultBackend}}
+		break
+	default:
+		return nil, fmt.Errorf("Unknown method, %s", method)
+	}
+
+	if err := retval.ParseRequest(request); err != nil {
+		return nil, err
+	}
+	return retval, nil
+}
+
+func (req *simpleJsonQueryRequest) GetReadRequest(session *pb.Session) *frames.ReadRequest {
+	if session == nil {
+		session = &pb.Session{Container: req.Container}
+	} else {
+		// don't overide the container (if one is already set)
+		if session.Container == "" {
+			session.Container = req.Container
+		}
+	}
+	return &frames.ReadRequest{Backend: req.Backend, Table: req.TableName, Columns: req.Fields,
+		Start: req.Range.From, End: req.Range.To, Step: "60m",
+		Session: session, Filter: req.Filter}
+}
+
+func (req *simpleJsonQueryRequest) formatKV(iter frames.FrameIterator) (interface{}, error) {
+	retVal := TableOutput{Type: "table", Rows: [][]interface{}{}}
+	for iter.Next() {
+		frame := iter.At()
+		iface, ok := frame.(pb.Framed)
+		if !ok {
+			return nil, errors.New("unknown frame type")
+		}
+		for _, column := range iface.Proto().Columns {
+			values, colType := req.extractColumnValues(column)
+			retVal.Columns = append(retVal.Columns, Column{Text: column.Name, Type: colType})
+			for rowIndex := 0; rowIndex < values.Len(); rowIndex++ {
+				if len(retVal.Rows) <= rowIndex {
+					retVal.Rows = append(retVal.Rows, []interface{}{})
+				}
+				value := values.Index(rowIndex).Interface()
+				retVal.Rows[rowIndex] = append(retVal.Rows[rowIndex], value)
+			}
+		}
+	}
+	if iter.Err() != nil {
+		return nil, iter.Err()
+	}
+	return []TableOutput{retVal}, nil
+}
+
+func (req *simpleJsonQueryRequest) formatTSDB(iter frames.FrameIterator) (interface{}, error) {
+	retval := []TimeSeriesOutput{}
+	for iter.Next() {
+		frame := iter.At()
+		iface, ok := frame.(pb.Framed)
+		if !ok {
+			return nil, errors.New("unknown frame type")
+		}
+		target := req.getTargetTSDB(iface.Proto())
+		indices := frame.Indices()
+		for _, column := range iface.Proto().Columns {
+			if column.Name == "values" {
+				datapoints := [][]interface{}{}
+				values, _ := req.extractColumnValues(column)
+				times, err := indices[0].Times()
+				if err != nil {
+					return nil, err
+				}
+				for j := 0; j < values.Len(); j++ {
+					datapoints = append(datapoints, []interface{}{values.Index(j).Interface(), req.formatTimeTSDB(times[j])})
+				}
+				retval = append(retval, TimeSeriesOutput{datapoints, target})
+			}
+		}
+	}
+	if iter.Err() != nil {
+		return nil, iter.Err()
+	}
+	return retval, nil
+}
+
+func (req *simpleJsonQueryRequest) extractColumnValues(column *pb.Column) (reflect.Value, string) {
+	colTypeStr := strings.ToLower(column.Dtype.String())
```

Comment:
Use extranal frame API here, the `Column` interface implements all that you want.